src/librbd/operation/DisableFeaturesRequest.cc

```diff
@@ -344,6 +342,8 @@ Context *DisableFeaturesRequest<I>::handle_disable_mirror_image(int *result) {
     lderr(cct) << "failed to disable image mirroring: " << cpp_strerror(*result)
                << dendl;
     // not fatal
+  } else if (m_mirror_mode == cls::rbd::MIRROR_MODE_IMAGE) {
+    m_enable_mirroring = true;
```

Comment:
If the mirror mode is journal and the pool mode is image, just prevent disabling the journal feature. Don't implicitly switch modes.